### PR TITLE
Support for local queries

### PR DESCRIPTION
**Pull request by jcs090218**
Feature for #94.

### CODE REVIEW

File: lisp/tree-sitter-hl.el [hunk 1]
```diff
@@ -143,6 +143,24 @@
...
+
+(defface tree-sitter-hl-face:local.definition
+  '((default :inherit font-lock-variable-name-face))
+  "Face for local definition."
+  :group 'tree-sitter-hl-faces)
+
+(defface tree-sitter-hl-face:local.reference
+  '((default :inherit font-lock-variable-name-face))
+  "Face for new local reference."
+  :group 'tree-sitter-hl-faces)
```
Comment (jcs090218) [inline lisp/tree-sitter-hl.el:162]: These are listed [here](https://github.com/tree-sitter/tree-sitter/blob/857a9ed07b983e2e6cff6dc851fcf9b37aec8e5a/docs/section-4-syntax-highlighting.md#local-variables).

File: langs/tree-sitter-langs-build.el [hunk 2]
```diff
@@ -353,28 +353,23 @@ non-nil."
...
-            (copy-file src dst-dir)))))))
+               file-name-as-directory (concat "queries"))))
+    (when (file-directory-p src)
+      (let ((dst-dir (file-name-as-directory
+                      (concat tree-sitter-langs--queries-dir
+                              (symbol-name lang-symbol)))))
+        (message "Copying queries for %s" lang-symbol)
+        (when (and force (file-directory-p dst-dir))
+          (delete-directory dst-dir t))
+        (copy-directory src dst-dir nil t t)))))
```
Comment (jcs090218) [inline langs/tree-sitter-langs-build.el:369]: Local query seems located in `locals.scm` and not `highlight.scm`.

Do you think we can just pull the entire query files down?

File: langs/tree-sitter-langs.el [hunk 3]
```diff
@@ -125,9 +123,12 @@ Return nil if there are no bundled patterns."
...
                              (_ nil))))
-          (insert-file-contents (tree-sitter-langs--hl-query-path sym))
-          (goto-char (point-max))
-          (insert "\n"))
+          (let* ((path (tree-sitter-langs--hl-query-path sym))
+                 (quries (directory-files path t "\\.scm$")))
+            (dolist (query quries)
+              (insert-file-contents query)
+              (goto-char (point-max))
+              (insert "\n"))))
```
Comment (jcs090218) [inline langs/tree-sitter-langs.el:131]: Load all quires at a time.

**Comment by ubolonton** [comment]
IIRC, local queries are used by upstream in a separate query pass, which then provides data for `.is-local?` predicates in highlighting queries. Concatenating them wouldn't work.

Did you get a working behavior instead?

**Comment by jcs090218** [comment]
Okay, sorry about this PR. I thought it would work in the first place. Although the highlighting did work but every thing seems to be highlighted as local variable too which is an unexpected behaviour. 😢 

I would like to implement this feature. Do you know where I should start?

**Comment by ubolonton** [comment]
> I would like to implement this feature. Do you know where I should start?

I just added some notes to #94.

**Pull request closed without merging**